tools/unicode_properties_parse/grapheme_break_test_data_gen.py

```diff
@@ -74,33 +73,33 @@ def parseBreakTestLine(input: TextIO) -> Optional[BreakTestItem]:
 
 def lineToCppDataLineUtf32(line: BreakTestItem) -> str:
     return cpp_test_data_line_template.format(','.join(
-        ["U'\\x" + format(x, 'x') + "'" for x in line.code_points]), ','.join(
+        [f"U'\\x{x:x}'" for x in line.code_points]), ','.join(
         [str(x) for x in line.breaks]))
 
 def lineToCppDataLineUtf8(line: BreakTestItem) -> str:
     utf8_rep = str(array('L', line.code_points),
                    encoding='utf-32').encode('utf-8')
     return cpp_test_data_line_template.format(','.join(
-        ["static_cast<char>(0x" + format(x, 'x') + ")" for x in utf8_rep]
+        [f"'\\x{x:x}'" for x in utf8_rep]
     ), ','.join([str(x) for x in line.breaks]))
 
 
 """
 Generate test data from "GraphemeBreakText.txt"
-This file can be downloaded from: https://www.unicode.org/Public/14.0.0/ucd/auxiliary/GraphemeBreakTest.txt.
+This file can be downloaded from: https://www.unicode.org/Public/14.0.0/ucd/auxiliary/GraphemeBreakTest.txt
 This script looks for GraphemeBreakTest.txt in same directory as this script
 """
 def generate_all() -> str:
     test_data_path = Path(__file__)
     test_data_path = test_data_path.absolute()
     test_data_path = test_data_path.with_name("GraphemeBreakTest.txt")
-    file = open(test_data_path, mode='rt', encoding='utf-8')
-    lines = list()
-    while line := parseBreakTestLine(file):
-        if len(line.code_points) > 0:
-            lines.append(line)
-    return cpp_template.format(len(lines), ','.join(map(lineToCppDataLineUtf32, lines)),
-          ','.join(map(lineToCppDataLineUtf8, lines)))
+    with open(test_data_path, mode='rt', encoding='utf-8') as file:
+        lines = list()
+        while line := parseBreakTestLine(file):
+            if len(line.code_points) > 0:
+                lines.append(line)
+        return cpp_template.format(len(lines), ','.join(map(lineToCppDataLineUtf32, lines)),

```

Comment:
nit: I don't think this has to be inside the with block. it doesn't really seem problematic to have it in there though.

Reply:
I did indeed unthinkingly indent everything. We can move `lines` out, and then move the `return` out.